src/linear_algebra/free_module.lean

```diff
@@ -155,3 +155,49 @@ of_basis (basis.of_vector_space R M)
 end division_ring
 
 end module.free
+
+/-! ### Standard inclusions
+We define in this section the standard inclusion `fin n → R →ₗ[R] (fin (n + m) → R)`.
+-/
+
+variables (ι : Type v) (η : Type w)
+
+/-- For any type `R` with `0`, we extend any `f : ι → R` to `(ι ⊕ η) → R`. -/
+def ext_zero [has_zero R] : (ι → R) → ((ι ⊕ η) → R) := λ f i, sum.cases_on i (λ j, f j) (λ j, 0)
```

Comment:
You need a better name here. `ext` means `extensionality`, not `extend`. The name also needs to have `sum` in it!

Reply:
Could you define this more generally for an injective map `f : X \to Y`? Make the new function zero on any point outside the range, and otherwise take the preimage. Not necessary to do in this generality, but might be nice.

Reply:
I just realized that we already have `function.extend`, so no need for a new definition.

Reply:
I have introduced `pi.extend_by_zero.linear_map` and friends, that work for any  injection of types.